molecule_openstack/cookiecutter/{{cookiecutter.molecule_directory}}/{{cookiecutter.scenario_name}}/molecule.yml

```diff
@@ -0,0 +1,16 @@
+---
+dependency:
+  name: {{ cookiecutter.dependency_name }}
+driver:
+  name: {{ cookiecutter.driver_name }}
+platforms:
+  - name: molecule-{{ cookiecutter.role_name }}
+    image: foo
+    flavor: bar
+    #network: define this if network is not molecule
```

Comment:
I wonder though that this still isn't ideal. It's much better, of course though! But it would be cool if these were just configured somehow without user intervention. The first `molecule test` will fail because these are still undefined. Could defaults be specified? Could the molecule API handle a way to pass these values in via the command line? Whatcha think?

Reply:
I think that even a hardcoded default that user needs to edit is better than not having any value at all. At this moment there is no support to override values but we should try to find a way to do it.

Reply:
The functionality of this is a bit entangled with https://github.com/ansible-community/molecule-openstack/pull/6 where you can also understand a bit the approach i took.
> But it would be cool if these were just configured somehow without user intervention.

I also think this is the ideal scenario by far. I find this a bit hard to implement on openstack as ( in comparison with aws where a lot of available resources are pre-defined ) every cluster can have a different set of available images/flavors/network setup etc. Therefore it's hard, if not impossible, to come up with sane defaults that should work in any setup. Based on that my approach was:
- Minimize any extra user configuration on molecule.yml: Therefore allow on molecule.yml to define any options that might be required (flavor, image, network, fip/no fip, ssh user)
-  This can make the molecule.yml a bit extended therefore introduce an extra level of "configuration" for "power users". E.g [here](https://github.com/ansible-community/molecule-openstack/blob/master/molecule_openstack/cookiecutter/%7B%7Bcookiecutter.molecule_directory%7D%7D/%7B%7Bcookiecutter.scenario_name%7D%7D/create.yml#L26) you can define the network to use and you don't have to define it on molecule.yml for every instance as i use it as [default](https://github.com/ansible-community/molecule-openstack/blob/master/molecule_openstack/cookiecutter/%7B%7Bcookiecutter.molecule_directory%7D%7D/%7B%7Bcookiecutter.scenario_name%7D%7D/create.yml#L78). Similar for ssh_user. ( I think for consistency reason we should intorduce the same mechanism for image and flavor or remove it from all options if we consider it as too much. Let me know what you think and i can make aPR for this. minal changes required)

---
>  Could the molecule API handle a way to pass these values in via the command line?

This is interesting. I am a bit skeptical though as to how much it will add. At the end is it easier to define 4 cli arguments or just adjust molecule.yml? For an automated process the first one in 10X easier but i'm just a bit skeptical in terms of what users will gain out of it.

---

Also as it is now, a user needs to define in any case the image, flavor and network to use as we don't create them. I also think that creating these resources can lead to too complicated set-up . E.g creating a new flavor is not hard but it might not work based on scheduling options of every cluster. Creating an image could work but it's a time expensive operation. Creating a network is easy but it wont work if it's not external and a fip is required, etc.  

Therefore, I think we should create some basic docs on how to configure molecule.yml to make it easier for new users. It might be good if we can link from the molecule driver section  docs to the molecule-openstack documentation. Also we might want to introduce a options check on `create.yml` that outputs a clear error msg when the input options are not sufficient.

Reply:
I noticed that `molecule 3.0.4` is released. I'll run some checks later in the evening for this PR to ensure it works as expected and i'll come back to you. The `TL;DR` of my previous comment is:
- I don't think its feasible to configure molecule-openstack in a way that 0 user interaction is needed, due to the nature of openstack
- Therefore I chose to have users only interract wth molecule.yml, and add some default mechanism on create.yml for "power users". This default mechanism is not in place for iamge and flavor, let me know if you want me to add it.
- I think we should add some simple documentation about this behavior to make it easier for new users. 

Let me know what you think

Reply:
Less interaction is likely better, no-interaction is a stretch goal which may never be fully achieved.

The image flavor and image name are good examples of stuff that prevents achieving it, as openstack has not defaults for that, at least in its current versions.

I would suggest adding an environment variable expansion like `MOLECULE_OS_IMAGE` and `MOLECULE_OS_FLAVOUR` which should help users figure a way to override them at runtime, w/o changing the molecule files itself.

Reply:
I like the env var idea. Should we then introduce this for all instance config options: 
`MOLECULE_OS_IMAGE => image`
`MOLECULE_OS_FLAVOR => flavor`
`MOLECULE_OS_SSH_USER => ssh_user`
`MOLECULE_OS_NETWORK => network`
`MOLECULE_OS_FIP_POOL => fip_pool`
Then the default `molecule.yml` will only define the instance name and the other will be left as comments. User can define the env vars which will be used across all instances, or populate `molecule.yaml`.

Reply:
Search for TEST_BASE_IMAGE on molecule code to see an example on how we use this feature. Maybe is would be better to use full driver name, just to make it less confusing, aka `MOLECULE_DRIVER_OPTION` kind of pattern. The best part is that you can still have a default value for the case when variable is not defined.

Reply:
Added variable substitution. rebased and squashed ( didn't want 4 commit for the same functionality)